P/PETSc/build_tarballs.jl

```diff
@@ -316,9 +365,24 @@ platforms = expand_gfortran_versions(supported_platforms(exclude=[Platform("i686
                                                                   Platform("armv6l","linux"; libc="musl"),
                                                                   Platform("armv7l","linux"; libc="musl"),
                                                                   Platform("armv7l","linux"; libc="gnu"),
-                                                                  Platform("aarch64","linux"; libc="musl")]))
-platforms, platform_dependencies = MPI.augment_platforms(platforms; MPItrampoline_compat=MPItrampoline_compat_version)
-
+                                                                  Platform("aarch64","linux"; libc="musl"),
+                                                                  Platform("aarch64","linux"; libc="musl"),
+                                                                  ]))
+
+# a few, but not all, platforms with libgfortran 3.0.0 are excluded
```

Comment:
I think it's easier to do filtering like https://github.com/JuliaPackaging/Yggdrasil/blob/7faddc5dca75e563154ad0cb5a1ab7ce84ed26fc/S/smesh/build_tarballs.jl#L37 Exact equality of platform objects is finicky, I don't recommend it. Basically you want to keep only freebsd and Windows with libgfortran 3? I fail to see a pattern here.